ocaml/xapi/xapi_db_upgrade.ml

```diff
@@ -235,23 +235,64 @@ let upgrade_wlb_configuration =
       )
   }
 
+(** On upgrade to the first ballooning-enabled XenServer, we reset memory
+    properties to safe defaults to avoid triggering something bad.
+    {ul
+    	{- For guest domains, we replace the current set of possibly-invalid memory
+    	constraints {i s} with a new set of valid and unballooned constraints {i t}
+    	such that:
+    	{ol
+    		{- t.dynamic_max := s.static_max}
+    		{- t.target      := s.static_max}
+    		{- t.dynamic_min := s.static_max}
+    		{- t.static_min  := minimum (s.static_min, s.static_max)}}}
+    	{- For control domains, we respect the administrator's choice of target:
+    	{ol
+    		{- t.dynamic_max := s.target}
+    		{- t.dynamic_min := s.target}}}
+    }
+*)
 let upgrade_vm_memory_for_dmc =
   {
-    description= "Upgrading VM memory fields to disable DMC"
-  ; version= (fun _ -> true)
+    description= "Upgrading VM memory fields for DMC"
+  ; version= (fun x -> x <= george)
```

Comment:
We can actually zap this entire rule now, since it is so old (George = XS 5.6 = 2009).

Reply:
I've removed this upgrade rule, and the unit test for it. Would you like to take a second look? Otherwise I will merge this.